pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -389,15 +415,79 @@ func (b *Base) Observables(rsrc interface{}, labels map[string]string) []reconci
 
 // Objects for Messaging service
 func (s *Messaging) Objects(rsrc interface{}, rsrclabels map[string]string, observed, dependent, aggregated []reconciler.Object) ([]reconciler.Object, error) {
-	r := rsrc.(*alpha1.CDAPMaster)
-	return getStatefulServiceResources(&r.Spec.Messaging.CDAPStatefulServiceSpec, rsrc, rsrclabels, alpha1.ServiceMessaging)
+	master := rsrc.(*alpha1.CDAPMaster)
```

Comment:
Why this logic in the object handler for the Messaging service? Shouldn't it be done in the Base object?